### PR TITLE
adds unary var matrix functions

### PR DESCRIPTION
**Pull request by SteveBronder**
## Summary

Adds `var<Matrix>` specializations for digamma, distance, Phi, Phi_approx, inv_Phi, sqrt, tail, and tgamma. 

## Tests

var matrix tests added for each function and can be run with 

```bash
./runTests.py test/unit/math/mix/fun/digamma_test.cpp \
test/unit/math/mix/fun/distance_test.cpp \
test/unit/math/mix/fun/inv_Phi_test.cpp \
test/unit/math/mix/fun/Phi_test.cpp \
test/unit/math/mix/fun/Phi_approx_test.cpp \
test/unit/math/mix/fun/sqrt_test.cpp \
test/unit/math/mix/fun/tail_test.cpp \
test/unit/math/mix/fun/tgamma_test.cpp
```

## Side Effects

Nope!

## Release notes

`var<Matrix>` overloads for digamma, distance, Phi, inv_Phi, Phi_approx, sqrt, tail, and tgamma

## Checklist

- [x] Math issue #1805 

- [x] Copyright holder: Steve Bronder

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

File: stan/math/rev/fun/Phi_approx.hpp [hunk 1]
```diff
@@ -44,12 +44,30 @@ namespace math {
...
   double da = f * (1 - f) * (3.0 * 0.07056 * av_squared + 1.5976);
-  return var(new precomp_v_vari(f, a.vi_, da));
+  return make_callback_var(
+      f, [a, da](auto& vi) mutable { a.adj() += vi.adj() * da; });
+}
+
+template <typename T, require_var_matrix_t<T>* = nullptr>
+inline auto Phi_approx(const T& a) {
+  arena_t<value_type_t<T>> f(a.rows(), a.cols());
+  arena_t<value_type_t<T>> da(a.rows(), a.cols());
```
Comment (rok-cesnovar) [inline stan/math/rev/fun/Phi_approx.hpp:57]: Does it make sense to compute this in the forward pass? I guess we would compute av_squared twice if we compute this in the backward pass. Is that the reason?

**Review by rok-cesnovar — COMMENTED** [review]
This looks good to me. I have one minor question. 

My first time reviewing varmat stuff so hopefully I didnt miss anything.

File: stan/math/rev/fun/Phi_approx.hpp (hunk 1 above)
Comment (SteveBronder) [reply stan/math/rev/fun/Phi_approx.hpp:57]: yeah that's the general jist. we can compute adjoint while calculating the forward pass so it's just nice to alloc everything at once, throw it in one big loop, and then have an easy peasy reverse pass.

Comment (rok-cesnovar) [reply stan/math/rev/fun/Phi_approx.hpp:57]: Cool.

File: stan/math/rev/fun/Phi.hpp [hunk 2]
```diff
@@ -62,7 +50,19 @@ class Phi_vari : public op_v_vari {
...
  * @return The unit normal cdf evaluated at the specified argument.
  */
-inline var Phi(const var& a) { return var(new internal::Phi_vari(a.vi_)); }
+inline var Phi(const var& a) {
+  return make_callback_var(Phi(a.val()), [a](auto& vi) mutable {
+    a.adj() += vi.adj() * INV_SQRT_TWO_PI * std::exp(-0.5 * a.val() * a.val());
+  });
+}
+
+template <typename T, require_var_matrix_t<T>* = nullptr>
```
Comment (rok-cesnovar) [inline stan/math/rev/fun/Phi.hpp:59]: We probably need doxygen here?

**Review by rok-cesnovar — CHANGES_REQUESTED** [review]
The varmat rev functions are missing doxygen. Other than that I think this looks good.

**Review by bbbales2 — DISMISSED** [review]
I added doxygen and simplified a couple tests. This looks good to me. @SteveBronder check my commits and merge if the tests pass.

**Review by rok-cesnovar — APPROVED** [review]
Good!

**Pull request merged**